searchlib/src/vespa/searchlib/attribute/attributeiterators.hpp

```diff
@@ -154,9 +212,23 @@ template <typename PL>
 void
 FilterAttributePostingListIteratorT<PL>::or_hits_into(BitVector & result, uint32_t begin_id) {
     (void) begin_id;
-    for (; _iterator.valid() && _iterator.getKey() < getEndId(); ++_iterator) {
-        if ( ! result.testBit(_iterator.getKey()) ) {
-            result.setBit(_iterator.getKey());
+    if constexpr (is_tree_iterator_v<PL>) {
+        auto end_itr = _iterator;
```

Comment:
This looks like the same code as on lines 163 - 173. Consider refactor into common function.